system-x/services/ldap/src/main/java/software/tnb/ldap/validation/LDAPValidation.java

```diff
@@ -0,0 +1,45 @@
+package software.tnb.ldap.validation;
+
+import software.tnb.common.validation.Validation;
+import software.tnb.ldap.account.LDAPAccount;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.unboundid.ldap.sdk.LDAPConnection;
+import com.unboundid.ldap.sdk.LDAPException;
+import com.unboundid.ldap.sdk.SearchRequest;
+import com.unboundid.ldap.sdk.SearchResult;
+import com.unboundid.ldap.sdk.SearchResultEntry;
+import com.unboundid.ldap.sdk.SearchScope;
+
+import java.util.List;
+
+public class LDAPValidation implements Validation {
+
+    private static final Logger LOG = LoggerFactory.getLogger(LDAPValidation.class);
+    private LDAPAccount ldapAccount;
+    private LDAPConnection ldapConnection;
+    private String ldapUrl;
+
+    public LDAPValidation(LDAPConnection ldapConnection, LDAPAccount ldapAccount, String ldapUrl) {
+        this.ldapConnection = ldapConnection;
+        this.ldapAccount = ldapAccount;
+        this.ldapUrl = ldapUrl;
+    }
+
+    public List<SearchResultEntry> getEntries(String filter) {
+        try {
+            ldapConnection.connect(ldapUrl.substring((ldapUrl.lastIndexOf('/') + 1), ldapUrl.lastIndexOf(':')),
+                Integer.parseInt(ldapUrl.substring(ldapUrl.lastIndexOf(':') + 1)));
+            ldapConnection.bind(ldapAccount.username(), ldapAccount.password());
```

Comment:
one last thing, the `connect` and `bind` should also be in the `openResources` method, same for redis `redisClient.connect();`

Reply:
I cannot do that for ldap, i need to `.connect()` and `.bind()` in validation for fresh connection to ldap. When the methods are in `openResources` the tests are failing because of not fresh connection.

For redis i need in the `RedisValidation` file `RedisClient` and `RedisConnection` at the same time. Is there a way how to do it simpler than it is now? Because when i move `redisClient.connect()` to the `openResources` i will lose the `RedisClient` variable.

Reply:
sorry, I didn't see you need both client and connection in the `RedisValidation`, so that can stay as it is now.

for LDAP it seems like it may be an issue with the client: https://github.com/pingidentity/ldapsdk/issues/131

as mentioned in the issue, you can use the `LDAPConnectionPool` instead of the `LDAPConnection` like this in the openResources method:

```
final LDAPConnection ldapConnection = new LDAPConnection();
ldapConnection.connect(StringUtils.substringBetween(url(), "ldap://", ":"), Integer.parseInt(StringUtils.substringAfterLast(url(), ':')));
ldapConnection.bind(account().username(), account().password());

client = new LDAPConnectionPool(ldapConnection, 1);
```

Reply:
Thanks for better solution, Done